aws/data_source_aws_route53_zone_test.go

```diff
@@ -25,13 +28,33 @@ func TestAccDataSourceAwsRoute53Zone(t *testing.T) {
 				Config: testAccDataSourceAwsRoute53ZoneConfig(rInt),
 				Check: resource.ComposeTestCheckFunc(
 					testAccDataSourceAwsRoute53ZoneCheck(
-						publicResourceName, "data.aws_route53_zone.by_zone_id", publicDomain),
+						publicResourceName,
+						"data.aws_route53_zone.by_zone_id",
+						map[string]string{"name": publicDomain},
+					),
 					testAccDataSourceAwsRoute53ZoneCheck(
-						publicResourceName, "data.aws_route53_zone.by_name", publicDomain),
+						publicResourceName,
+						"data.aws_route53_zone.by_name",
+						map[string]string{"name": publicDomain},
+					),
 					testAccDataSourceAwsRoute53ZoneCheck(
-						privateResourceName, "data.aws_route53_zone.by_vpc", privateDomain),
+						privateResourceName,
+						"data.aws_route53_zone.by_vpc",
+						map[string]string{"name": privateDomain},
+					),
 					testAccDataSourceAwsRoute53ZoneCheck(
-						privateResourceName, "data.aws_route53_zone.by_tag", privateDomain),
+						privateResourceName,
+						"data.aws_route53_zone.by_tag",
+						map[string]string{"name": privateDomain},
+					),
+					testAccDataSourceAwsRoute53ZoneCheck(
```

Comment:
Since this acceptance testing was originally written, the Terraform Provider SDK provides a much easier way to verify state attribute values match between two resources via [`resource.TestCheckResourceAttrPair()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPair) and values within a single resource via [`resource.TestCheckResourceAttr()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttr) and [`resource.TestMatchResourceAttr()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestMatchResourceAttr).

Instead of changing the `testAccDataSourceAwsRoute53ZoneCheck` function to further increase its complexity, we would prefer to start improving the testing to use the simpler functions instead of the these legacy methods if possible.

e.g. in this case:

```go
resource.TestCheckResourceAttrPair("data.aws_route53_zone.service_discovery_by_vpc", "name", serviceDiscoveryResourceName, "name"),
resource.TestCheckResourceAttr("data.aws_route53_zone.service_discovery_by_vpc", "linked_service_principal", "servicediscovery.amazonaws.com"),
// if the value matches an attribute value in the resource, use TestCheckResourceAttrPair() instead of the below
resource.TestMatchResourceAttr("data.aws_route53_zone.service_discovery_by_vpc", "linked_service_description", regexp.MustCompile(`^arn:[^:]+:servicediscovery:[^:]+:[^:]+:namespace/ns-\\w+$`)),
```